### PR TITLE
Moves all logical expressions to ternary conditions

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR moves all logical expressions to ternary conditions to avoid these kind of errors.
It also updates the mobile eslint config.

### Related PR:

https://github.com/CoorpAcademy/eslint-plugin-coorpacademy/pull/40

### Testing strategy

- [x] I ensured my code is tested
- [x] I took care to mention breaking changes and extra libs if appropriate: 
-> `@coorpacademy/eslint-plugin-coorpacademy@9.2.1` (from 8.3.1)
-> `babel-eslint@10.0.3` (from 8.2.6)
-> `eslint@6.8.0` (from 5.16.0)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -70,6 +70,18 @@
...
+    "react/jsx-curly-newline": "off",
+    "react/state-in-constructor": "off",
+    "react/jsx-one-expression-per-line": "off",
+    "default-param-last": "off",
+    "unicorn/consistent-function-scoping": "off",
+    "unicorn/no-unused-properties": "off",
+    "prefer-regex-literals": "off",
+    "no-useless-catch": "off",
+    "react/jsx-no-useless-fragment": "off",
+    "unicorn/no-console-spaces": "off"
```
Comment (VincentCATILLON) [inline .eslintrc]: should be fixed

File: package.json [hunk 2]
```diff
@@ -132,7 +132,7 @@
     "@babel/core": "^7.1.5",
     "@babel/node": "^7.0.0",
     "@babel/runtime": "^7.4.4",
-    "@coorpacademy/eslint-plugin-coorpacademy": "^8.3.1",
+    "@coorpacademy/eslint-plugin-coorpacademy": "9.2.0-alpha.0",
```
Comment (VincentCATILLON) [inline package.json]: temporary

File: package.json [hunk 3]
```diff
@@ -143,17 +143,14 @@
...
     "babel-plugin-module-resolver": "^3.1.1",
     "babel-plugin-transform-inline-environment-variables": "^0.4.3",
     "babel-plugin-transform-remove-console": "^6.9.4",
     "detox": "14.8.2",
-    "eslint": "^5.5.0",
+    "eslint": "^6.7",
     "eslint-import-resolver-babel-module": "^5.0.0-beta.1",
-    "eslint-plugin-flowtype": "^2.50.0",
-    "eslint-plugin-import": "^2.14.0",
-    "eslint-plugin-react": "^7.11.1",
```
Comment (VincentCATILLON) [inline package.json]: these deps were added but not directly required

**Comment by Fausto95** [comment]
LGTM, just fix the tests pls

**Comment by VincentCATILLON** [comment]
Missing thing:
- Fix components / screens coverage

**Comment by VincentCATILLON** [comment]
@Fausto95 should be fixed now :)

**Comment by VincentCATILLON** [comment]
@Fausto95 done :)

File: package.json [hunk 4]
```diff
@@ -143,17 +143,14 @@
...
     "babel-core": "^7.0.0-bridge.0",
-    "babel-eslint": "^9.0.0",
+    "babel-eslint": "^10.0.3",
     "babel-jest": "^23.6.0",
     "babel-plugin-module-resolver": "^3.1.1",
     "babel-plugin-transform-inline-environment-variables": "^0.4.3",
     "babel-plugin-transform-remove-console": "^6.9.4",
     "detox": "14.8.2",
-    "eslint": "^5.5.0",
+    "eslint": "^6.7",
```
Comment (Fausto95) [inline package.json]: .7 tout court?

Comment (VincentCATILLON) [reply package.json]: y :)

**Pull request merged**